### PR TITLE
chore(githubAction): add build workflow initial work for argocd move

### PR DESCRIPTION
**Pull request by Basma1912**
## One-line summary

- [ ] I used an AI to write some of this code.

### CODE REVIEW

**Comment by pmac** [comment]
It looks fine to me. @stevejalim you mind taking a look?

**Comment by Basma1912** [comment]
I just added the prod runner to the workflow.

**Review by stevejalim — APPROVED** [review]
r+wc

Looks good to me - thanks @Basma1912 !

I have a couple of queries, all just to be sure I understand it

1) Am I right that we'll be using eactly the same image for dev, stage and prod, just adding a new tag as we move from main -> stage -> prod? 

Put another way, on a push to `main`, we'll get an image created and tagged with  `dev-abcd123`, then when we push to stage, the image will already exist in GAR and will just have the `stage-abcd123` tag added, and then a similar think for prod (with a different style of label). Is my take right? I just want to be sure that we can still be sure that the same container image used on dev/stage is what we ship to prod. (I think it will be)

2) Am I right that there'll be an event/trigger set in ArgoCD to start running our integration tests against a particular environment once the rollout to that environment is complete?

Thanks for your work on this! ArgoCD seems great

**Comment by Basma1912** [comment]
@stevejalim sorry I missed the mention , looking at [this](https://github.com/mozilla-sre-deploy/deploy-bedrock/blob/main/event-router.yaml)  

push to main  -> build an image for DEV env 
push to run-integration-tests - > build an image for test env 
push to stage - > build an image for stage 
push tags - > build an image for prod  

[ the above is what currently in place for bedrock ] 

looking at the event router .. I don't have any context whether all the changes in main eventually get pushed to stage and finally got tagged and deployed to prod 

if all branches are kept in sync .. then yes .. stage and dev will be running the same image .. [regardless the build process ran twice ] 

if you want to change the current workflow .. and deploy to dev and stage when a push is done to the same branch .. we can do so . 

hope this answers your question :)

**Comment by stevejalim** [comment]
> if you want to change the current workflow 

Oh, no need to change it - if we retain what we have now, that's great. Thanks!

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  GAR_LOCATION: us
+  GCP_PROJECT_ID: moz-fx-bedrock-prod
+  GAR_REPOSITORY: bedrock
+  REF_ID: ${{ github.ref }}
+
+
+jobs:
+  build_and_publish_public_images:
+    name: Build public Bedrock images and push to Docker hub
+    runs-on: [self-hosted, nonprod, prod, us-central1, bedrock]
```
Comment (bkochendorfer) [inline .github/workflows/build-and-push.yml:32]: Do we want this to run on both `nonprod` and `prod`? Or is there some case somewhere that puts the `prod` build on `prod` and `nonprod` on `nonprod`?

**Comment by Basma1912** [comment]
you are right no need to include both .. since we are only using the workflow to build image .. I'll keep prod runner only ..is that ok?

**Comment by bkochendorfer** [comment]
Yep sounds good

**Comment by stevejalim** [comment]
@Basma1912 Done! 👍

**Pull request merged**